src/modules/data/PreImageInfo.ts

```diff
@@ -21,41 +21,53 @@ import { Validator, IPreImageInfo } from './validator'
  */
 export class PreImageInfo
 {
+    /**
+     * UTXO used to enroll
+     */
     public enroll_key: Hash;
+
+    /**
+     * Value of the pre-image
+     */
     public hash: Hash;
-    public distance: number = 0;
 
-    constructor (enroll_key?: Hash, hash?: Hash, distance?: number)
+    /**
+     * Distance from the enrollment, 0 based
+     */
+    public distance: number;
+
+    /**
+     * Construct a new instance of this object
+     *
+     * @param enroll_key The UTXO used to enroll
+     * @param hash       The value of the pre-image
+     * @param distance   The distance from the Enrollment
+     */
+    constructor (enroll_key: Hash, hash: Hash, distance: number)
     {
-        if (enroll_key !== undefined)
-            this.enroll_key = enroll_key;
-        else
-            this.enroll_key = new Hash();
-
-        if (hash !== undefined)
-            this.hash = hash;
-        else
-            this.hash = new Hash();
-
-        if (distance !== undefined)
-            this.distance = distance;
-        else
-            this.distance = 0;
+        this.enroll_key = enroll_key;
+        this.hash = hash;
+        this.distance = distance;
     }
 
     /**
-     * This parses JSON.
-     * @param json The object of the JSON
-     * @returns The instance of PreImageInfo
+     * The reviver parameter to give to `JSON.parse`
+     *
+     * This function allows to perform any necessary conversion,
+     * as well as validation of the final object.
+     *
+     * @param key   Name of the field being parsed
+     * @param value The value associated with `key`
+     * @returns A new instance of `PreImageInfo` if `key == ""`, `value` otherwise.
      */
-    public parseJSON (json: any): PreImageInfo
+    public static reviver (key: string, value: any): any
     {
-        Validator.isValidOtherwiseThrow<IPreImageInfo>('PreImageInfo', json);
+        if (key !== "")
```

Comment:
Do you allow whitespaces in the empty string check method?

Reply:
No, there's no reason to do so.